src/openrct2/cmdline/BenchLogic.cpp

```diff
@@ -0,0 +1,136 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2019 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#include "CommandLine.hpp"
+
+#ifdef USE_BENCHMARK
+
+#    include "../Context.h"
+#    include "../Game.h"
+#    include "../Intro.h"
+#    include "../OpenRCT2.h"
+#    include "../audio/audio.h"
+#    include "../core/Console.hpp"
+#    include "../core/Imaging.h"
+#    include "../drawing/Drawing.h"
+#    include "../interface/Viewport.h"
+#    include "../localisation/Localisation.h"
+#    include "../paint/Paint.h"
+#    include "../platform/platform.h"
+#    include "../util/Util.h"
+#    include "../world/Climate.h"
+#    include "../world/Map.h"
+#    include "../world/Park.h"
+#    include "../world/Surface.h"
+
+#    include <benchmark/benchmark.h>
+#    include <cstdint>
+#    include <iterator>
+#    include <vector>
+
+#include "../network/network.h"
+#include "../world/Sprite.h"
+#include "../GameState.h"
+
+
+#include <cstdlib>
+#include <chrono>
+#include <memory>
+
+using namespace OpenRCT2;
+
+static void BM_gamelogic(benchmark::State& state, IContext* context)
```

Comment:
Why can't it use the standard form
```
for (auto _ : state)
{
    context->GetGameState()->UpdateLogic();
}
```
?

Reply:
Actually, you found the problem with my code, you actually need this. Why?

Reply:
The loop is actually measuring the time, you don't need your timer. I'm just not sure it is safe to call `context->GetGameState()->UpdateLogic()` repeatedly in a loop.